File: 0011-container-with-most-water/0011-container-with-most-water.go
func maxArea(height []int) int {
    res := 0
    left, right := 0, len(height) - 1

    for left < right {
        cur := (right - left) * min(height[left], height[right])
        if cur > res {
            res = cur
        }

        if height[left] < height[right] {
            left++
        } else {
            right--
        }
    }

    return res
}

func min(x, y int) int {
    if x < y {
        return x
    }
    return y
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height) -> int:
        left, right = 0, len(height)-1
        res = 0

        while left < right:
            cur = (right - left) * min(height[left], height[right])
            res = max(res, cur)

            if height[left] < height[right]:
                left += 1
            else:
                right -= 1

        return res
